M2/Macaulay2/packages/Complexes/ChainComplexMap.m2

```diff
@@ -99,13 +99,9 @@ map(Complex, Complex, ZZ) := ComplexMap => opts -> (D, C, j) -> (
         result.cache.isCommutative = true;
         return result
         );
-    if j === 1 then (
-        if C == D and (opts.Degree === null or opts.Degree === 0) then
-            return id_C;
-        error "expected source and target to be the same";
-        );
-    error "expected integer to be zero or one";
-    )
+    if C == D and (opts.Degree === null or opts.Degree === 0) then
+        return j * id_C;
+    error "expected 0 or source and target to be the same")
```

Comment:
@mahrud Why is this change to `map` desirable?  Doesn't this functionality already exist with `j * id_C`?

Reply:
I don't remember how exactly where, but I think we wanted the same function to work both for modules and complexes, so we changed this to conform with this module morphism syntax:
```m2
i1 : M = module(QQ[x,y]);

i2 : map(M, M, 2)

o2 = | 2 |

                      1               1
o2 : Matrix (QQ[x..y])  <-- (QQ[x..y])
```
Since this method exists already, isn't it simpler not to single out `j == 1` specifically In particular, modules also allow this:
```m2
i6 : map(M, M, x)

o6 = | x |

                      1               1
o6 : Matrix (QQ[x..y])  <-- (QQ[x..y])
```
But this would be an additional method, and I agree with you that `x * id_C` is simpler in this case.